emscripten.py

```diff
@@ -814,16 +812,17 @@ def create_module(receiving, invoke_funcs, metadata, library_symbols):
     module.append("var asm = createWasm();\n")
 
   module.append(receiving)
-  module.append(invoke_wrappers)
+  if not settings.DISABLE_EXCEPTION_CATCHING or settings.SUPPORT_LONGJMP == 'emscripten':
+    module.append(create_invoke_wrappers(metadata))
```

Comment:
How about adding and `else` here with `assert not metadata.invokeFuncs, "invoke_ functions exported but exceptions and longjmp are both disabled"`

Reply:
b7e19a9592c5e1eaa2b0e760e2853892cbd1c4e4

Reply:
Can you flip the two parts of this condition so that negative one comes second?  (Avoids having to know that `not` has a higher precedence than `or` then reading the code).

Reply:
98c79d70b136ef97322a3278e7d2072091c08304